app/controllers/api/internal/v1/laa_references_controller.rb

```diff
@@ -26,6 +28,25 @@ def allowed_params
         def transformed_params
           create_params.slice(*allowed_params).to_hash.transform_keys(&:to_sym)
         end
+
+        def make_common_platform_calls(contract)
+          maat_reference = contract[:maat_reference]
+          defendant_id = contract[:defendant_id]
+          offences = ProsecutionCaseDefendantOffence.where(defendant_id: defendant_id)
+          prosecution_case_id = offences.pluck(:prosecution_case_id).first
```

Comment:
since `ProsecutionCaseDefendantOffence` contain all the required info, perhaps we could loop through the `offences` itself , as `pluck` might be making extra db calls.
eg: 
```
ProsecutionCaseDefendantOffence.where(defendant_id: defendant_id).each do |case|
    LaaReferenceUpdaterJob.call(case.offence_id, case.prosecution_case_id, ....
    ...
end
```

Reply:
Yes, that's neater. Thanks.